File: ReactNative/appreveal/ios/ScreenInfo.swift
// Screen identity model

import Foundation

/// Information about the currently active screen.
public struct ScreenInfo: Codable {
    public let screenKey: String
    public let screenTitle: String
    public let frameworkType: String // "uikit", "swiftui"
    public let controllerChain: [String]
    public let activeTab: String?
    public let navigationDepth: Int
    public let presentedModals: [String]
    public let confidence: Double
    /// How the screen was identified: "explicit", "derived"
    public let source: String
    /// Title extracted from navigation bar
    public let appBarTitle: String?
}

File: ReactNative/appreveal/android/src/main/kotlin/com/appreveal/screen/ScreenInfo.kt
package com.appreveal.screen

/**
 * Information about the currently active screen.
 * Mirrors the iOS ScreenInfo struct exactly.
 */
data class ScreenInfo(
    val screenKey: String,
    val screenTitle: String,
    val frameworkType: String,         // "android", "compose", "react-native", "unknown"
    val activityChain: List<String>,   // equivalent of iOS controllerChain
    val activeTab: String?,
    val navigationDepth: Int,
    val presentedModals: List<String>,
    val confidence: Double,
    /** How the screen was identified: "explicit", "derived" */
    val source: String,
    /** Title extracted from Toolbar/ActionBar */
    val appBarTitle: String?
)
